src/generate/api.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { model } from "amf-client-js";
+
+import {
+  resolveApiModel,
+  parseRamlFile,
+  getAllDataTypes
+} from "../common/parser";
+
+import Name from "./name";
+
+/**
+ * An API represented as an AMF model. Includes the extracted data types
+ * defined in the spec. Common transformations of the name are cached for
+ * reference in templates and for use in assembling file paths.
+ */
+export default class Api {
+  private _dataTypes: model.domain.CustomDomainProperty[] = [];
+  private _model = new model.document.Document();
+  private _name = new Name();
+  private _path = "";
```

Comment:
Why are these private variables with getters?

Reply:
All of the fields here are somewhere in the model. I can't come up with a good reason to not keep them in sync.